includes/classes/Indexable/Comment/Comment.php

```diff
@@ -809,23 +827,33 @@ public function query_db( $args ) {
 		unset( $all_query_args['offset'] );
 		$all_query_args['count'] = true;
 
-		/**
-		 * Filter database arguments for comment count query
-		 *
-		 * @hook ep_comment_all_query_db_args
-		 * @param  {array} $args Query arguments based to WP_Comment_Query
-		 * @since  3.6.0
-		 * @return {array} New arguments
-		 */
-		$total_objects = get_comments( apply_filters( 'ep_comment_all_query_db_args', $all_query_args, $args ) );
-
-		if ( ! empty( $args['offset'] ) ) {
-			if ( (int) $args['offset'] >= $total_objects ) {
-				$total_objects = 0;
-			}
+		if ( isset( $args['comment__in'] ) || 0 < $args['offset'] ) {
+			// Disable advanced pagination. Not useful if only indexing specific IDs.
+			$args['ep_indexing_advanced_pagination'] = false;
 		}
 
-		$query = new WP_Comment_Query( $args );
+		// Enforce the following query args during advanced pagination to ensure things work correctly.
+		if ( $args['ep_indexing_advanced_pagination'] ) {
+			$args = array_merge(
+				$args,
+				[
+					'suppress_filters' => false,
+					'orderby'          => 'ID',
+					'order'            => 'DESC',
+					'paged'            => 1,
+					'offset'           => 0,
+				]
+			);
+			add_filter( 'comments_clauses', array( $this, 'bulk_indexing_filter_comments_where' ), 9999, 2 );
```

Comment:
```suggestion
			add_filter( 'comments_clauses', [ $this, 'bulk_indexing_filter_comments_where' ], 9999, 2 );
```
(we should likely enforce the use of [] instead of array() in our CS)